File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        sort = sorted(nums)
        i, j = 0, 1

        while j < len(sort):
            if sort[i] == sort[j]:
                return sort[i]
            i += 1
            j += 1
        return -1

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> answer = new HashMap<>();

        for (String str : strs) {
            char[] array = str.toCharArray();
            Arrays.sort(array);
            String sorted = new String(array);

            answer.putIfAbsent(sorted, new ArrayList<>());
            answer.get(sorted).add(str);
        }

        return new ArrayList<>(answer.values());
    }
}
